### ISSUE TITLE
Occasional pauses while typing, during which keystrokes are lost

### ISSUE THREAD

**Issue by HighCommander4** [tomboy-notes/tomboy#82]
When using Tomboy intensively for a period of time (like an hour or more), I notice that occasionally it pauses for a few seconds, during which time it's not responsive to keystrokes.

I'm not sure what the pauses are correlated with - it writing the notes to disk, or perhaps it synchronizing them (I do have synchronization enabled).

Now, being unresponsive for a few seconds would be fine, if, as in other applications, the keystrokes are buffered up and processed once the pause is over. However, what I'm experiencing is that keystrokes typed during the pause are not processed at all - they are lost.

It would be great if this could be fixed, so that the keystrokes aren't lost. It seems like it should be technically possible, as every other editor I've used does it (buffers keystrokes during a UI hang and processes them afterwards).

**Comment by davidbannon** [comment]
Nathan, I'm not offering to try and address this issue for you, sorry. But I hope you might help me (with tomboy-ng) a little by providing some extra info ?
Can you tell me how many notes you have and how big they are ?  If you are using Linux, perhaps 
ls -l ~/.local/share/tomboy | wc -l
and 
du -hs ~/.local/share/tomboy/*

My guess is your problem relates to the size of your notes. My view is, with tomboy-ng at least, the long term solution is to move the notes into a lightweight database and avoid the repetitive i/o. As far as buffering Tomboy's keystrokes, I'm surprised it does not do that, I suspect it could be a significant change I'm afraid. 
Anyone know if Tomboy is multithreaded ?
It would interesting to see if turning off auto sync for awhile makes the problem go away.

**Comment by HighCommander4** [comment]
> Can you tell me how many notes you have and how big they are ?

I have 11 notes and their total size on disk is 1.1 MB.

> It would interesting to see if turning off auto sync for awhile makes the problem go away.

Good idea; I will try that and report back.

What is tomboy-ng?

**Comment by davidbannon** [comment]
OK, 11 is not many at all but each note is seriously large !  I'd say 'document' rather than 'note' :-) I'm quite surprised, my biggest note is 45K, your average is 100K. You have given me reason to think where I need to take tomboy-ng, thanks !

Reasonably modern hardware ?

tomboy-ng is a move to build a new version of Tomboy solving some of the cross platform and maintenance issues. See https://wiki.gnome.org/Apps/Tomboy/tomboy-ng

If you have a working copy of Tomboy, unless you are looking to help with the project (and that would be great!), you are probably better off where you are for the time being.

David

**Comment by HighCommander4** [comment]
> OK, 11 is not many at all but each note is seriously large ! I'd say 'document' rather than 'note' :-) I'm quite surprised, my biggest note is 45K, your average is 100K.

Yeah, I use it to take notes during conferences. They can get pretty detailed, and I put all my notes for a given conference (~1 week long) into one note.

> Reasonably modern hardware ?

Yep (core i7 processor, 8 GB RAM, SSD storage).

**Comment by alex-ter** [comment]
You can try running it with the -"--debug" switch, which woukd output additional info onto console (and log file) and see if there's anythng that correlates with stalls.
